### PR TITLE
Implement TextualEvent tiles for im.vector.modular.widgets

### PR DESCRIPTION
**Pull request by lukebarnard1**
E.g. "Bob added a Acme widget", "Susan removed a Giraffe widget"  The name is calculated by taking the `name` in the event content, falling back on the `type`, falling back on the previous content `type`. This is then capitalised.

Should be squash merged.

### CODE REVIEW

File: src/TextForEvent.js [hunk 1]
```diff
@@ -248,6 +248,26 @@ function textForPowerEvent(event) {
     });
 }
 
+function textForWidgetEvent(event) {
+    const senderName = event.sender ? event.sender.name : event.getSender();
+    const previousContent = event.getPrevContent() ? event.getPrevContent() : {};
```
Comment (superdump) [inline src/TextForEvent.js]: I don't know how costly a called to `getPrevContent()` is, but in JavaScript I think the idiom is:

```
const previousContent = event.getPrevContent() || {};
```

File: src/TextForEvent.js [hunk 2]
```diff
@@ -248,6 +248,26 @@ function textForPowerEvent(event) {
...
+    const {name, type, url} = event.getContent() ? event.getContent() : {};
```
Comment (superdump) [inline src/TextForEvent.js]: Same here: `const {name, type, url} = event.getContent() || {};`

File: src/TextForEvent.js [hunk 3]
```diff
@@ -248,6 +248,26 @@ function textForPowerEvent(event) {
...
+    let widgetName = widgetName || name || type || previousContent.type || '';
+
+    // Apply sentence case
+    widgetName = widgetName ? widgetName[0].toUpperCase() + widgetName.slice(1).toLowerCase() + ' ' : '';
```
Comment (superdump) [inline src/TextForEvent.js]: Is there any good reason to lower-case the rest of the string? I think maybe preserving the case of the rest of the string is better. That is, just upper-case the first character and leave the rest as it is.

File: src/TextForEvent.js [hunk 4]
```diff
@@ -248,6 +248,26 @@ function textForPowerEvent(event) {
...
+
+    if (url) {
```
Comment (superdump) [inline src/TextForEvent.js]: Technically if the widget state event has no content, then it has been removed, but I think this is sufficiently equivalent as having a url is required.

File: src/TextForEvent.js (hunk 1 above)
Comment (lukebarnard1) [inline src/TextForEvent.js]: oh. absolutely

File: src/TextForEvent.js (hunk 3 above)
Comment (lukebarnard1) [inline src/TextForEvent.js]: This is the same logic applied to the app tile header, I was going for consistency. Ideally the name would be formatted as intended for human reading by Scalar itself.

File: src/TextForEvent.js (hunk 4 above)
Comment (lukebarnard1) [inline src/TextForEvent.js]: I shall comment this, thanks

File: src/TextForEvent.js (hunk 3 above)
Comment (superdump) [inline src/TextForEvent.js]: OK. Fine by me.

**Comment by superdump** [comment]
LGTM.

**Pull request merged**